kornia/utils/helpers.py

```diff
@@ -0,0 +1,24 @@
+from typing import Tuple, Union, List, cast, Optional
+
+import torch
+
+
+def _extract_device_dtype(tensor_list: List[Optional[torch.Tensor]]):
+    """This function will check if all the input tensors are in the same device.
+
+    If so, it would return a tuple of (device, dtype)
+    """
+    device, dtype = None, None
+    for tensor in tensor_list:
+        if tensor is not None:
+            if not isinstance(tensor, (torch.Tensor,)):
+                raise ValueError(f"Expected None or Tensor. Got {tensor}.")
+            _device = tensor.device
+            _dtype = tensor.dtype
+            if device is None and dtype is None:
+                device = _device
+                dtype = _dtype
+            elif device != _device or dtype != _dtype:
+                raise ValueError("Passed values are not in the same device and dtype."
+                                 f"Got ({device}, {dtype}) and ({_device}, {_dtype}).")
+    return (device, dtype)
```

Comment:
@shijianjian this should have  minimal unit test

Reply:
couple of comments, no typing in the output and input should be `List[torch.Tensor] -> Tuple[torch.device, torch.dtype]`